test/e2e/README.md

```diff
@@ -0,0 +1,7 @@
+# Test
+
+## Compatibility notice
+
+This package is not subject to deprecation notices or compatibility guarantees.
+
+- Breaking changes are likely. External providers using this package should update to the latest API changes when updating Cluster API Operator. Maintainers and contributors must give notice in release notes when a breaking change happens.
```

Comment:
👍 
we can always set depreciation annotation on methods, IDEs and linters will complain in that case